engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -129,6 +134,29 @@ class NextgenEngine @JvmOverloads constructor(
         return result
     }
 
+    private suspend fun resolveServiceAndExecuteTopLevelField(
+        field: ExecutableNormalizedField,
+        executionContext: NadelExecutionContext,
+        service: Service
+    ): ExecutionResult {
+        return if (usesDynamicServiceResolution(topLevelField = field)) {
```

Comment:
Is it possible to move this code into `NadelFieldToService`. In my mind it belongs there. We have the ability to raise exceptions that implement `GraphQLError` to achieve the same effect.

Reply:
Yep sounds good. I was worried that I was adding too much code in `NextgenEngine` thanks for the suggestion!